### PR TITLE
feat: add nft playground

### PR DESCRIPTION
**Pull request by alessey**
**What changed? Why?**
* Added playground, including reservoir data hooks
* Added /mint route to playground to support earnings
* Added specific data hook for this route without the owners api call

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: playground/nextjs-app-router/package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "@radix-ui/react-switch": "^1.1.0",
     "@radix-ui/react-tabs": "^1.1.0",
     "@rainbow-me/rainbowkit": "^2.1.3",
+    "@reservoir0x/reservoir-sdk": "^2.4.25",
```
Comment (alessey) [inline playground/nextjs-app-router/package.json:27]: using this to type the data returned from reservoir in the playground hooks

File: src/nft/components/NFTCard.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            border.radius,
+            'flex w-full max-w-[500px] flex-col items-stretch border p-4 text-left',
+            `hover:border-[${border.defaultActive}]`,
+            className,
+          )}
+          data-testid="ockNFTCard_Container"
+          onClick={handleOnClick}
+        >
+          {children}
+        </button>
```
Comment (alessey) [inline src/nft/components/NFTCard.tsx:64]: NFT Card is now wrapped as a button that links to opensea

File: src/nft/components/mint/NFTMintButton.tsx [hunk 3]
```diff
@@ -74,6 +85,7 @@ export function NFTMintButton({
   return (
     <div className={cn('py-2', className)}>
       <Transaction
+        capabilities={capabilities}
```
Comment (alessey) [inline src/nft/components/mint/NFTMintButton.tsx:88]: adding paymaster

File: src/styles/theme.ts [hunk 4]
```diff
@@ -13,22 +13,23 @@ export const text = {
...
+    'cursor-pointer ock-bg-inverse active:bg-[var(--ock-bg-inverse-active)] hover:bg-[var(--ock-bg-inverse-hover)]',
   primary:
-    'cursor-pointer ock-bg-primary active:ock-bg-primary-active hover:bg-[var(--ock-bg-primary-hover)]',
+    'cursor-pointer ock-bg-primary active:bg-[var(--ock-bg-primary-active)] hover:bg-[var(--ock-bg-primary-hover)]',
   secondary:
-    'cursor-pointer ock-bg-secondary active:ock-bg-secondary-active hover:bg-[var(--ock-bg-secondary-hover)]',
+    'cursor-pointer ock-bg-secondary active:bg-[var(--ock-bg-secondary-active)] hover:bg-[var(--ock-bg-secondary-hover)]',
   coinbaseBranding:
-    'cursor-pointer bg-[#0052FF] active:ock-bg-secondary-active hover:bg-[#0045D8]',
+    'cursor-pointer bg-[#0052FF] active:bg-[var(--ock-bg-secondary-active)] hover:bg-[#0045D8]',
```
Comment (alessey) [inline src/styles/theme.ts:32]: these active styles were not previously working

File: playground/nextjs-app-router/components/demo/NFTMintCard.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+// autumn 0x473d2D4C09669962c2CbDB1c34ba8f0fc843Fb69
+// fall at the florida zoo 0x8443d418e82c3A2A530013218D49E0347BA13fDe ERC1155
+// some other base mint 0x96c2E93Aa994D0A467f377c77075B00421EFB046  ERC1155
+// bored ape gang 0x381b0d693dc6c5ea556838e8604ae000510a16ac 1
+// persona 0x6502820f3f035c7a9fc0ebd3d74a0383306c5137
+// mochimons 0x949bed087ff0241e04e98d807de3c3dd97eaa381
+// hoomans 0xbe3c7abab76f0a1de602fdb2f44faf604a5f649f
+// mr miggles video 0x1f52841279fA4dE8B606a70373E9c84e84Ce9204
+// cb quarterly earnings 0x1D6b183bD47F914F9f1d3208EDCF8BefD7F84E63
+// audio - 0x05a28e3d5f68c8b4a521ab7f74bd887fae6a598d base song
```
Comment (cpcramer) [inline playground/nextjs-app-router/components/demo/NFTMintCard.tsx:32]: Same as above - share in a const file

File: playground/nextjs-app-router/lib/useReservoirMintData.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+          headers: {
+            accept: '*/*',
+            'x-api-key': ENVIRONMENT_VARIABLES.RESERVOIR_API_KEY ?? '',
+          },
+        },
+      );
+      const data = await response.json();
+      // if no tokenId, get the collection and default to the first token
+      return data.tokens[0].token as Token;
+    },
```
Comment (abcrane123) [inline playground/nextjs-app-router/lib/useReservoirMintData.ts:35]: little confused by this comment. should we first try to grab by tokenId before grabbing the first token?

File: playground/nextjs-app-router/lib/useReservoirNFTData.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+          method: 'GET',
+          headers: {
+            accept: '*/*',
+            'x-api-key': ENVIRONMENT_VARIABLES.RESERVOIR_API_KEY ?? '',
+          },
+        },
+      );
+      const data = await response.json();
+      // if no tokenId, get the collection and default to the first token
+      return data.tokens[0].token as Token;
```
Comment (abcrane123) [inline playground/nextjs-app-router/lib/useReservoirNFTData.ts:30]: same with this comment

File: src/nft/components/NFTMintCard.tsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  onSuccess,
+}: NFTMintCardReact) {
+  const componentTheme = useTheme();
+
+  const isMounted = useIsMounted();
+
+  // prevents SSR hydration issue
+  if (!isMounted) {
+    return null;
+  }
```
Comment (abcrane123) [inline src/nft/components/NFTMintCard.tsx:26]: i'm not familar with the order of these components but do we need the `isMounted` in both this component and `NFTCard`?

File: playground/nextjs-app-router/lib/useReservoirMintData.ts (hunk 6 above)
Comment (alessey) [reply playground/nextjs-app-router/lib/useReservoirMintData.ts:35]: this endpoint is set up to accept tokens=ContractAddress:TokenId or collection=ContractAddress.  If no tokenId is passed in, I call with collection which returns all the tokens in the collection, and I just pick the first one.  It'd most likely be the same as calling with tokenId = 0, but also would handle weird contracts that started at 1

File: playground/nextjs-app-router/lib/useReservoirNFTData.ts (hunk 7 above)
Comment (alessey) [reply playground/nextjs-app-router/lib/useReservoirNFTData.ts:30]: yea, this is the same hook.  I can refactor into nicer code

File: playground/nextjs-app-router/app/mint/page.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
+'use client';
```
Comment (abcrane123) [inline playground/nextjs-app-router/app/mint/page.tsx:1]: why do we need a separate page for this component?

File: playground/nextjs-app-router/app/mint/page.tsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+        schemaId="0xf8b05c79f090979bf4a80270aba232dff11a10d9ca55c4f88de95317970f0de9"
+      >
+        <NFTMintCard
+          contractAddress="0x1D6b183bD47F914F9f1d3208EDCF8BefD7F84E63"
+          tokenId="3"
+          useNFTData={useEarningsData}
+          buildMintTransaction={buildMintTransaction}
+          onStatus={(status: LifecycleStatus) => {
+            console.log('status', status);
+          }}
```
Comment (abcrane123) [inline playground/nextjs-app-router/app/mint/page.tsx:46]: nit but can we move this to cb function defined above to keep consistent with other components

File: playground/nextjs-app-router/app/mint/page.tsx (hunk 9 above)
Comment (alessey) [reply playground/nextjs-app-router/app/mint/page.tsx:1]: I'm hoping to use this page in an iframe for the earnings mint published at onchainkit.xyz/playground/mint

File: playground/nextjs-app-router/app/mint/page.tsx [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+import { NFTMedia } from '@coinbase/onchainkit/nft/view';
+import { base } from 'wagmi/chains';
+import { useEarningsData } from './hooks/useEarningsData';
+
+// const SUCCESS_URL =
+//   'https://investor.coinbase.com/events-and-presentations/events/event-details/2024/Third-Quarter-2024-Earnings-Call/default.aspx';
+
+// const handleSuccessClick = useCallback(() => {
+//   window.open(SUCCESS_URL, '_blank', 'noopener,noreferrer');
+// }, []);
```
[the remark this answers is not in the document]
Comment (alessey) [reply playground/nextjs-app-router/app/mint/page.tsx:22]: I'll remove this for now.  I need to update the component to integrate your overrides and then I can implement them here.

File: playground/nextjs-app-router/app/mint/page.tsx (hunk 9 above)
Comment (0xAlec) [reply playground/nextjs-app-router/app/mint/page.tsx:1]: can we add a custom app logo and name here? the keys.coinbase.com popup will say `OnchainKit` with a null logo currently, the earnings team will likely want their own customization

Comment (0xAlec) [reply playground/nextjs-app-router/app/mint/page.tsx:1]: you may need to override the `WagmiProvider` since it's controlled by the `config` value there

**Review by 0xAlec — APPROVED** [review]

**Review by abcrane123 — APPROVED** [review]

**Pull request merged**